### ISSUE TITLE
🚧Setup multiple OS runs

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/Infrastructure#7]
### I have done the items below . . .

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Set up the build and test workflows to take an optional input that can be used to tell the workflow to run **Windows**,  run **Ubuntu**.  The input should be optional with a default value of empty.  If it is empty, it will run **Ubuntu**.

This will play an important role in the future when building and testing multiple operating systems.

To pull this off, you will have to use `if:` logic in the YAML to test the input value.
Example: You will have to create 2 jobs.  One for **Windows** and one for **Ubuntu**.

To avoid case-sensitive values causing issues, you will have to use some PowerShell to set the values of the inputs to lowercase values for non-case sensitive comparisons in the `if:` logic.

Refer to the example below:
```yml
# Windows Only
my_windows_job:
    name: My Windows Job
    if: ${{ inputs.os == 'windows' }}
    runs-on: windows-latest
    steps:
    - uses: actions/checkout@v2

# Ubuntu Only
my_ubuntu_job:
    name: My Ubuntu Job
    if: ${{ inputs.os == 'ubuntu' }}
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
```

Do this for the build and test jobs.  Also make sure to update the workflow names.
Currently the name of the build is **✔Build Status Check**, rename it to **Build Project**

### Acceptance Criteria

- [ ] New input named `os` added
  - [ ] `os` input set to required false of type `string`
- [ ] Build jobs updated
  - [ ] **Windows** job created
    - Runs only if `os` input is set to `windows`
  - [ ] **Ubuntu** job created
    - Runs only if `os` input is set to `ubuntu`
- [ ] Unit test jobs updated
  - [ ] **Windows** job created
    - Runs only if `os` input is set to `windows`
  - [ ] **Ubuntu** job created
    - Runs only if `os` input is set to `ubuntu`

### ToDo Items

- [ ] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [ ] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [ ] Issue linked to the correct project.
- [ ] Draft pull request created and linked to this issue.

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                                                                                      |
|---------------------|--------------------------------------------------------------------------------------------|
| Bug Fixes           | https://github.com/KinsonDigital/Infrastructure/labels/%F0%9F%90%9Bbug                     |
| Breaking Changes    | https://github.com/KinsonDigital/Infrastructure/labels/%F0%9F%A7%A8breaking%20changes      |
| Enhancement         | https://github.com/KinsonDigital/Infrastructure/labels/enhancement                         |
| Workflow Changes    | https://github.com/KinsonDigital/Infrastructure/labels/workflow                            |
| Code Doc Changes    | https://github.com/KinsonDigital/Infrastructure/labels/%F0%9F%93%91documentation%2Fcode    |
| Product Doc Changes | https://github.com/KinsonDigital/Infrastructure/labels/%F0%9F%93%9Ddocumentation%2Fproduct |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label                                                                    |
|---------------------|--------------------------------------------------------------------------|
| Low Priority        | https://github.com/KinsonDigital/Infrastructure/labels/low%20priority    |
| Medium Priority     | https://github.com/KinsonDigital/Infrastructure/labels/medium%20priority |
| High Priority       | https://github.com/KinsonDigital/Infrastructure/labels/high%20priority   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by CalvinWilkinson** [comment]
Put on hold.

This is because currently, a new build system is being built and this will not be required anymore.

Once the build system is complete, the status can be updated and then closed without implementation

**Comment by CalvinWilkinson** [comment]
Issue closed.

**Reason:** The new build system CICD is finished and being used.
